frmts/saga/sagadataset.cpp

```diff
@@ -304,6 +306,18 @@ double SAGARasterBand::GetNoDataValue( int * pbSuccess )
     return m_NoData;
 }
 
+/************************************************************************/
+/*                           SetNoDataValue()                           */
+/************************************************************************/
+
+CPLErr SAGARasterBand::SetNoDataValue(double dfNoData)
+{
+    m_NoData = dfNoData;
+    SAGADataset * poSAGADS = static_cast<SAGADataset *>(poDS);
+    poSAGADS->headerDirty=1;
```

Comment:
```suggestion
    poSAGADS->headerDirty=true;
```
Please also poSAGADS->eAccess == GA_Update at the beginning (similar to what is done in SetGeoTransform())

The CreateCopy() implementation also needs to call that method

Reply:
> The no data value of the raster band is already copied in CreateCopy
ok, I missed that it was setting it through the creation option (I thought that calling SetNoDataValue() would be now needed). So disregard that part of my comment

Reply:
It is also a bit weird that the projection is written when calling SetProjection . Perhaps this should also toggle the headerDirty bit and be saved only when closing the dataset to make it consistent with header changes?

Reply:
> It is also a bit weird that the projection is written when calling SetProjection . Perhaps this should also toggle the headerDirty bit and be saved only when closing the dataset to make it consistent with header changes?

up to you. As it writes another file than the .sgrd header, this isn't strictly needed (technically if defered writing of the .prj file is done, this should be done through another flag that headerDirty)

Reply:
Okay, I'll leave it as is in that case.